betree/pmdk/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+
+            dest_filehandle = match libpmem::pmem_file_create(&DEST_FILEPATH, 64*1024*1024*1024, &mut mapped_len, &mut is_pmem) {
+                        Some(existing_handle) => existing_handle,
+                        None => match libpmem::pmem_file_open(&DEST_FILEPATH, &mut mapped_len, &mut is_pmem) {
+                            Some(new_handle) => new_handle,
+                            None => panic!("\n Failed to create or open pmem file handle.")
+                        }
+                    };
+
+
+            //dest_filehandle = libpmem::pmem_file_open(&DEST_FILEPATH, &mut mapped_len, &mut is_pmem);
+            //dest_filehandle = libpmem::pmem_file_create(&DEST_FILEPATH, 16*1024*1024, &mut mapped_len, &mut is_pmem);
+
+            let mut text_array = [0u8; 4096];
+            TEXT.bytes().zip(text_array.iter_mut()).for_each(|(b, ptr)| *ptr = b);
+            libpmem::pmem_file_write(&dest_filehandle, 0, &text_array, TEXT.chars().count());
+
+            TEXT2.bytes().zip(text_array.iter_mut()).for_each(|(b, ptr)| *ptr = b);
+            libpmem::pmem_file_write(&dest_filehandle, TEXT.chars().count(), &text_array, TEXT2.chars().count());
+
+            let mut buffer = vec![0; TEXT.chars().count()];
+            libpmem::pmem_file_read(&dest_filehandle, 0, &mut buffer, TEXT.chars().count() as u64);
+            println!("\n TEXT length: {}", TEXT.chars().count());
+
+            let mut buffer2 = vec![0; TEXT2.chars().count()];
+            libpmem::pmem_file_read(&dest_filehandle, TEXT.chars().count(), &mut buffer2, TEXT2.chars().count() as u64);
+            println!("\n TEXT2 length: {}", TEXT2.chars().count());
+
+            TEXT.bytes().zip(text_array.iter_mut()).for_each(|(b, ptr)| *ptr = b);
+            libpmem::pmem_file_write(&dest_filehandle, 1000, &text_array, TEXT.chars().count());
+
+            let mut buffer3 = vec![0; TEXT.chars().count()];
+            libpmem::pmem_file_read(&dest_filehandle, 1000, &mut buffer3, TEXT.chars().count() as u64);
+            println!("\n TEXT length: {}", TEXT.chars().count());
+
+
+            libpmem::pmem_file_close(&dest_filehandle, &mapped_len);
+
+            let read_string = match std::str::from_utf8(&buffer) {
+                Ok(string) => string,
+                Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+            };
+
+            assert_eq!(TEXT, read_string);
+            println!("{}", read_string);
+
+            let read_string2 = match std::str::from_utf8(&buffer2) {
+                Ok(string) => string,
+                Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+            };
+
+            assert_eq!(TEXT2, read_string2);
+
+
+            let read_string3 = match std::str::from_utf8(&buffer3) {
+                Ok(string) => string,
+                Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+            };
+
+            assert_eq!(TEXT, read_string3);            
+        }
+    }
+}
```

Comment:
Move from test to example. We cannot guarantee that machines which want to test the code have access to NVM e.g. CI.

Reply:
We have added the feature 'nvm' to exclude the compilation of NVM-related code for such scenarios, right?